documentation/docs/pages/operators/nodes/performance-and-testing/node-api-check.mdx

```diff
@@ -0,0 +1,160 @@
+```sh
+# for http
+http://<PUBLIC_IP>:8080/api/v1/roles
+# or
+http://<PUBLIC_IP>/api/v1/roles
+
+# for reversed proxy/WSS
+https://<HOSTNAME>/api/v1/roles
+```
+
+## `node_api_check.py`
+
+To make this a bit easier, we made a CLI tool querying all available API endpoints based on node `Identity Key` (further denoted as `<ID_KEY>`) called `node_api_check.py`. To diagnose your node performance, whether by yourself or by sharing an output in our [operator channel](https://matrix.to/#/#operators:nymtech.chat), this tool provides you with a quick overview of live data. We recommend to run this checker alongside [`nym_gateway_probe`](gateway-probe.md) to triage both performance and an actual routing.
+
+Besides querying any bonded node APIs, `nym_api_check.py` has a function counting all existing nodes in provided version.
+
+### Setup
+
+#### Pre-requsities
+
+<Steps>
+
+###### 1. Install and configure Python3
+
+
+- Start with installing Python3:
+```sh
+sudo apt install python3
+```
+
+- Make sure Python3 is your default Python version:
+```sh
+update-alternatives --install  /usr/bin/python python /usr/bin/python3 1
+```
+
+- Verify:
+```sh
+python --version
+
+# should return higher than 3
+```
+
+- Install Python modules `tabulate`, `pandas` and `argparse` either using [`pip`](https://python.land/virtual-environments/installing-packages-with-pip) or if you installed Python3 system-wide you can install modules directly:
+
+<div>
+  <Tabs items={[
+    <strong>Using <code>pip</code></strong>,
+    <strong>System-wide Python installation</strong>,
+    ]} defaultIndex="1">
+    <MyTab>
+      ```sh
+      pip install tabulate pandas argparse
+      ```
+    </MyTab>
+    <MyTab>
+      ```sh
+      sudo apt install python3-tabulate python3-pandas python3-argparse -y
+      ```
+    </MyTab>
+  </Tabs>
+</div>
+
+###### 2. Install `node_api_check.py` and make executable  
+
+To run the program you neet to have  [`node_api_check.py`](https://github.com/nymtech/nym/tree/develop/scripts/node_api_check.py) and [`api_endpoints.json`](https://github.com/nymtech/nym/tree/develop/scripts/api_endpoints.json). 
+
+- If you [compiled from source](../../binaries/building-nym.mdx), you already have both of these files. Note that the latest version of this program may be on `develop` branch.
+
+- If you prefer to download them individually, do it by opening terminal in your desired location and running:
+```sh
+wget https://raw.githubusercontent.com/nymtech/nym/tree/develop/node_api_check.py
+
+wget https://raw.githubusercontent.com/nymtech/nym/tree/develop/api_endpoints.json
+```
+
+- Make executable:
+```sh
+chmod u+x node_api_check.py
+```
+
+</Steps>
+
+Now you are ready to check your node.
+
+### Usage & Commands
+
+- Run the program with `--help` flag to see the available commands:
+```sh
+./node_api_check.py --help
+```
+
+- Output:
+```python
+<!--cmdrun cd ../../../../scripts && python ./node_api_check.py --help-->
```

Comment:
also remove?

Reply:
I left all cmdrun for easier location post merge